partner_phonecall_schedule/models/res_partner.py

```diff
@@ -47,24 +49,20 @@ def _compute_phonecall_calendar_ids(self):
 
     def _search_phonecall_available(self, operator, value):
         """Search quickly if partner is available to call right now."""
-        Attendance = self.env["resource.calendar.attendance"]
-        available = Attendance.search(
+        attendance = self.env["resource.calendar.attendance"]
+        available = attendance.search(
             self._phonecall_available_domain(),
         )
         if operator == "!=" or "not" in operator:
             value = not value
         operator = "in" if value else "not in"
         return [("phonecall_calendar_ids.attendance_ids",
-                operator, available.ids)]
+                 operator, available.ids)]
 
     def _phonecall_available_domain(self):
         """Get a domain to know if we are available to call a partner."""
-        now = self.env.context.get("now", datetime.now())
-        try:
-            now = fields.Datetime.from_string(now)
-        except TypeError:
-            # `now` is already a datetime object
-            pass
+        now = fields.Datetime.from_string(self.env.context.get(
```

Comment:
Use directly `datetime` Python type, and use `fields.Datetime.now()`.

Reply:
If we will use `fields.Datetime.now()` then UT will be failed due to  `self.assertTrue(dude.phonecall_available)`

Reply:
2018-12-25 04:53:34,633 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: FAIL: test_friday_morning (odoo.addons.partner_phonecall_schedule.tests.test_schedule.CanICallCase) 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: ` I can call dude this morning 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: Traceback (most recent call last): 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 138, in test_friday_morning 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self.allowed() 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 94, in allowed 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self._allowed() 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `   File "/home/anand/workspace/oca/anand/partner-contact/partner_phonecall_schedule/tests/test_schedule.py", line 71, in _allowed 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: `     self.assertTrue(dude.phonecall_available) 
2018-12-25 04:53:34,634 5987 ERROR test_partner_phonecall_schedule_12 odoo.addons.partner_phonecall_schedule.tests.test_schedule: ` AssertionError: False is not true

Reply:
OK